skglm/solvers/prox_newton.py

```diff
@@ -0,0 +1,338 @@
+import numpy as np
+from numba import njit
+from scipy.sparse import issparse
+
+
+EPS_TOL = 0.3
+MAX_CD_ITER = 20
+MAX_BACKTRACK_ITER = 20
+
+
+def prox_newton(X, y, datafit, penalty, w_init=None, p0=10,
+                max_iter=20, max_epochs=1000, tol=1e-4, verbose=0):
+    """Run a Prox Newton solver.
```

Comment:
I may help to Add refs to the blitz and celer (JMLR) papers and codebases. You can put (see References at this line, and a reference section at the end).

Reply:
also specify that it is combined with working sets